uniflow/op/model/lm/model_server.py

```diff
@@ -262,81 +264,66 @@ class HuggingfaceModelServer(AbsModelServer):
-                max_new_tokens=self._model_config.max_new_tokens,
-                num_return_sequences=self._model_config.num_return_sequences,
-                repetition_penalty=self._model_config.repetition_penalty,
-                eos_token_id=tokenizer.eos_token_id,
-                pad_token_id=tokenizer.pad_token_id,
-                batch_size=self._model_config.batch_size,
-            )
+        self._model_config = HuggingfaceModelConfig(**model_config)
+
+        if self._model_config.model_name == "google/gemma-7b-it":
+            self._initialize_gemma_model()
         else:
-            if self._model_config.load_in_4bit or self._model_config.load_in_8bit:
-                self._model_config.load_in_4bit = False
-                self._model_config.load_in_8bit = False
-                print(
-                    "Neuron model does not support quantized models. load_in_4bit and load_in_8bit are automatically set to False."
-                )
-            from uniflow.op.model.lm.neuron_utils import (  # pylint: disable=import-outside-toplevel
-                Neuron,
-            )
+            self._initialize_default_model()
 
-            model, tokenizer = Neuron.get_neuron_model(
-                self._model_config.model_name, self._model_config.batch_size
-            )
-            self._pipeline = partial(
-                Neuron.neuron_infer,
-                model=model,
-                tokenizer=tokenizer,
-                max_new_tokens=self._model_config.max_new_tokens,
-                batch_size=self._model_config.batch_size,
-            )
-        self._tokenizer = tokenizer
+    def _initialize_gemma_model(self):
+        self._tokenizer, self._model = self._get_model(is_gemma=True)
+        self._pipeline = self._create_pipeline(self._model, self._tokenizer)
 
-    def _get_model(self):
-        """Get model."""
-        from transformers import (  # pylint: disable=import-outside-toplevel
-            AutoModelForCausalLM,
-            AutoTokenizer,
-        )
+    def _initialize_default_model(self):
+        try:
+            self._tokenizer, self._model = self._get_model()
+            self._pipeline = self._create_pipeline(self._model, self._tokenizer)
+        except ModuleNotFoundError as exc:
+            raise ModuleNotFoundError(
+                "Please install transformers to use HuggingfaceModelServer."
+            ) from exc
 
+    def _get_model(self, is_gemma: bool = False):
+        """Get model."""
         tokenizer = AutoTokenizer.from_pretrained(
             self._model_config.model_name,
+            torch_dtype="auto" if is_gemma else None,
+            device_map="auto",
+            use_auth_token=self._model_config.token if is_gemma else None,
         )
         tokenizer.pad_token = tokenizer.eos_token
         model = AutoModelForCausalLM.from_pretrained(
             self._model_config.model_name,
+            torch_dtype="auto" if is_gemma else None,
```

Comment:
why we need this? Does Gemma not support torch_dtype? If I recall, I assume Gemma has a torch https://github.com/google/gemma_pytorch implementation.